src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -306,6 +316,11 @@ private String indexNameFor(final long suffix) {
         return String.format(Locale.ROOT, "%s.%s.%d", IP2GEO_DATA_INDEX_NAME_PREFIX, id, suffix);
     }
 
+    /**
+     * Tell if current datasource is expired or not
```

Comment:
nit: may be Checks if datasource is expired or not

Reply:
Done